python/lancedb/table.py

```diff
@@ -500,6 +501,65 @@ def add(
         lance.write_dataset(data, self._dataset_uri, schema=self.schema, mode=mode)
         self._reset_dataset()
 
+    def merge(
+        self,
+        other_table: Union[LanceTable, ReaderLike],
+        left_on: str,
+        right_on: Optional[str] = None,
+        schema: Optional[pa.Schema, LanceModel] = None,
+    ):
+        """Merge another table into this table.
+
+        Performs a left join, where the dataset is the left side and data_obj
+        is the right side. Rows existing in the dataset but not on the left will
+        be filled with null values, unless Lance doesn't support null values for
```

Comment:
What happens if a row in the dataset matches multiple rows in `other_table`?  In a typical `left join` I would expect to get one output row per match but I assume that can't happen here (since we aren't actually rewriting the left side).

Reply:
We collect the right side into a hashmap, so the values on the RHS are de-depulicated.